src/Microsoft.OData.Core/UriParser/TypePromotionUtils.cs

```diff
@@ -493,8 +493,20 @@ internal static bool CanConvertTo(SingleValueNode sourceNodeOrNull, IEdmTypeRefe
                 return true;
             }
 
+			// We must support conversion from/to untyped.
+			if (sourceReference.IsUntyped() || targetReference.IsUntyped())

```

Comment:
>|| [](start = 35, length = 2)

|| or && ?

Reply:
Should be ||. We should allow conversions of untyped to/from anything. The actual instance values may fail to convert, but the expression should be allowed.

---
In reply to: [320886093](https://github.com/OData/odata.net/pull/1489#discussion_r320886093) [](ancestors = 320886093)